### PR TITLE
Enhance packaging phase by embedding catalog entries

### PR DESCRIPTION
**Pull request by boubaker**
Before, the addons had to be declared in remote catalog in order to release a version or when creating a new Feature Branch.
With this change, a local catalog is generated at build/release time in order to determine addons versions and URLs to be pre-bundled.

### CODE REVIEW

File: packaging/plf-community-tomcat-standalone/src/main/catalog/meeds-distrib-catalog.json [hunk 1]
```diff
@@ -0,0 +1,115 @@
+[
+  {
+    "id": "meeds-app-center",
+    "version": "${addon.exo.app-center.version}",
+    "unstable": true,
+    "name": "App Center",
+    "description": "An addon to manage displayed user applications in menu",
+    "sourceUrl": "https://github.com/meeds-io/app-center",
+    "downloadUrl": "http://addons.exoplatform.org/n/exo-addons-snapshots/org.exoplatform.addons.app-center/app-center-packaging/${addon.exo.app-center.version}/zip",
```
Comment (ghost) [inline packaging/plf-community-tomcat-standalone/src/main/catalog/meeds-distrib-catalog.json]: These links will work only for snapshot version, Other versions ( milestone will be impacted )

Comment (boubaker) [reply packaging/plf-community-tomcat-standalone/src/main/catalog/meeds-distrib-catalog.json]: Ah, yes, good point @hbenali . Which permanent URL to use ?

Comment (boubaker) [reply packaging/plf-community-tomcat-standalone/src/main/catalog/meeds-distrib-catalog.json]: Is it of format:
` https://repository.exoplatform.org/service/local/artifact/maven/redirect?r=exo-jboss-snapshots&g=org.exoplatform.jcr&a=jcr-packaging&v=${addon.exo.jcr.version}&e=zip`
I think the name of repository is mandatory here as well, isn't it ?

Comment (boubaker) [reply packaging/plf-community-tomcat-standalone/src/main/catalog/meeds-distrib-catalog.json]: Ah, found it, we can use `public` proxy repository. I will test it, but will it access to `staging` repository too ?

Comment (boubaker) [reply packaging/plf-community-tomcat-standalone/src/main/catalog/meeds-distrib-catalog.json]: Fixed cc @hbenali

**Review by bourasom — DISMISSED** [review]
LGTM

**Review by ghost — DISMISSED** [review]

**Review by ahamdi — APPROVED** [review]

**Review by rdenarie — APPROVED** [review]

**Pull request merged**